### PR TITLE
Use repository's name when possible instead of local directory name

### PR DESCRIPTION
**Pull request by danielkalen**
Addresses https://github.com/desktop/desktop/issues/1701 and https://github.com/desktop/desktop/issues/664

### CODE REVIEW

File: app/src/models/repository.ts [hunk 1]
```diff
@@ -59,6 +59,7 @@ export class Repository implements IRepository {
   }
 
   public get name(): string {
-    return path.basename(this.path)
+    const repoName = this.gitHubRepository && this.gitHubRepository.name
```
Comment (shiftkey) [inline app/src/models/repository.ts]: Given `this.path` and `this.githubRepository` are set in the constructor, and how much we use this, we should compute this there and cache it

**Comment by danielkalen** [comment]
Since I accidentaly messed up the commits across the branches in this fork I created a new branch, applied the requested changes, and created a new PR here https://github.com/desktop/desktop/pull/1801

I'm afraid I will need some pointers on where to start with the tests as this project is relatively massive and I'm no TypeScript expert... Analyzing the project's test suite on my own in attempt to figure it out will take me a lot of time.

**Comment by shiftkey** [comment]
>I'm afraid I will need some pointers on where to start with the tests as this project is relatively massive and I'm no TypeScript expert... Analyzing the project's test suite on my own in attempt to figure it out will take me a lot of time.

@danielkalen :thumbsup: I'll carve out some time early next week for this and see if I can turn it into some proper documentation for the project.

**Comment by shiftkey** [comment]
> Since I accidentaly messed up the commits across the branches in this fork I created a new branch, applied the requested changes, and created a new PR here

@danielkalen for future reference, if things get really bad with a branch on your fork you can just force-push (`git push origin {branch} -f` )your changes and they'll get synchronized to the PR.

**Comment by danielkalen** [comment]
>  I'll carve out some time early next week for this and see if I can turn it into some proper documentation for the project.
@shiftkey sounds great! I'll be awaiting your feedback. If you find your self spending too much time on this I won't mind dealing with a very quick rundown on where to start with the test suites.

> for future reference, if things get really bad with a branch on your fork you can just force-push (git push origin {branch} -f )your changes and they'll get synchronized to the PR.
Hmm... well the issue for me wasn't really to push to remote after rebasing, but rather the issue was avoiding a merge that git was forcing me to perform which I wasn't able to get past - therefore I created a new branch. As for this PR I noticed that I accidentally used the master branch of the fork instead of a dedicated branch.

**Pull request closed without merging**